homeassistant/components/devialet/media_player.py

```diff
@@ -0,0 +1,215 @@
+"""Support for Devialet speakers."""
+from __future__ import annotations
+
+from devialet.const import NORMAL_INPUTS
+
+from homeassistant.components.media_player import (
+    MediaPlayerEntity,
+    MediaPlayerEntityFeature,
+    MediaPlayerState,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, MANUFACTURER, SOUND_MODES
+from .coordinator import DevialetCoordinator
+
+SUPPORT_DEVIALET = (
+    MediaPlayerEntityFeature.VOLUME_SET
+    | MediaPlayerEntityFeature.VOLUME_MUTE
+    | MediaPlayerEntityFeature.TURN_OFF
+    | MediaPlayerEntityFeature.SELECT_SOURCE
+    | MediaPlayerEntityFeature.SELECT_SOUND_MODE
+)
+
+DEVIALET_TO_HA_FEATURE_MAP = {
+    "play": MediaPlayerEntityFeature.PLAY | MediaPlayerEntityFeature.STOP,
+    "pause": MediaPlayerEntityFeature.PAUSE,
+    "previous": MediaPlayerEntityFeature.PREVIOUS_TRACK,
+    "next": MediaPlayerEntityFeature.NEXT_TRACK,
+    "seek": MediaPlayerEntityFeature.SEEK,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Devialet entry."""
+    client = hass.data[DOMAIN][entry.entry_id]
+    coordinator = DevialetCoordinator(hass, client)
+    await coordinator.async_config_entry_first_refresh()
+
+    async_add_entities([DevialetMediaPlayerEntity(coordinator, entry)])
+
+
+class DevialetMediaPlayerEntity(CoordinatorEntity, MediaPlayerEntity):
+    """Devialet media player."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    def __init__(self, coordinator, entry: ConfigEntry) -> None:
+        """Initialize the Devialet device."""
+        self.coordinator = coordinator
+        super().__init__(coordinator)
+
+        self._attr_unique_id = (
+            entry.unique_id
+            if entry.unique_id is not None
+            else coordinator.client.serial
```

Comment:
Why is this else needed, when is a config entry without a unique_id created?

Reply:
Without the else this mypy error occurs: 

`homeassistant/components/devialet/media_player.py:62: error: Argument 1 to <set> has incompatible type "tuple[str, str | None]"; expected "tuple[str, str]"  [arg-type]
Found 1 error in 1 file (checked 1 source file)`

Reply:
I see, but we should not work around typing issues by changing the code.
Instead, do something like this:
```py
assert entry.unique_id
self._attr_unique_id = entry.unique_id
```

Reply:
Casting fixed this indeed. Thanks!